cmd/release/github.go

```diff
@@ -0,0 +1,168 @@
+)
+
+func getClient(accessToken string) *github.Client {
+	ctx, cancel := context.WithTimeout(context.Background(), contextTimeout)
+	defer cancel()
+
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{
+			AccessToken: accessToken,
+		},
+	)
+
+	client := github.NewClient(oauth2.NewClient(ctx, ts))
+
+	return client
+}
+
+func new(accessToken string) *app {
+	return &app{
+		owner:  "MeilleursAgents",
+		repo:   "terraform-provider-ansiblevault",
+		client: getClient(accessToken),
+	}
+}
+
+func (a *app) PublishArtifactsOnTag(tag string, buildAssets []string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), contextTimeout)
+	defer cancel()
+
+	log.Println("==> Retrieve release informations from tag ...")
+	release, err := a.retrieveTag(ctx, tag)
+
+	if err != nil {
+		return errors.Wrapf(err, "[ERROR] When retrieve release information %s", release.GetHTMLURL())
+	}
+
+	log.Println("==> Retrieve release assets ...")
+	releaseRemoteAssets, err := a.listAssets(ctx, release.GetID())
+
+	if err != nil {
+		return errors.Wrapf(err, "[ERROR] When retrieve assets %s", release.GetHTMLURL())
+	}
+
+	assetsToUpload := a.computeUploadAssets(releaseRemoteAssets, buildAssets)
+
+	if len(assetsToUpload) > 0 {
+		log.Println("==> Upload release assets ...")
+		err = a.uploadAssets(ctx, release.GetID(), assetsToUpload)
+
+		if err != nil {
+			return errors.Wrapf(err, "[ERROR] uploadAssets on %s with %+q", release.GetHTMLURL(), assetsToUpload)
+		}
+	} else {
+		log.Println("==> Nothing to upload ...")
+	}
+
+	return nil
+}
+
+func (a *app) computeUploadAssets(releaseRemoteAssets []*github.ReleaseAsset, buildAssets []string) []string {
+	remoteAssetsNotFound := []string{}
+	remoteAssets := []string{}
+
+	for _, value := range releaseRemoteAssets {
+		remoteAssets = append(remoteAssets, *value.Name)
+	}
+
+	for _, value := range buildAssets {
+		if !contains(remoteAssets, strings.Replace(value, "dist/", "", -1)) {
+			remoteAssetsNotFound = append(remoteAssetsNotFound, value)
+		}
+	}
+
+	return remoteAssetsNotFound
+}
+
+func (a *app) retrieveTag(ctx context.Context, tag string) (*github.RepositoryRelease, error) {
+	release, res, err := a.client.Repositories.GetReleaseByTag(ctx, a.owner, a.repo, tag)
+	if err != nil {
+		return nil, err
+	}
+
+	if res.StatusCode != http.StatusOK {
+		return nil, errors.Wrapf(err, "invalid status code: %s", res.Status)
+	}
+
+	return release, nil
+}
+
+func (a *app) listAssets(ctx context.Context, releaseID int64) ([]*github.ReleaseAsset, error) {
+	result := []*github.ReleaseAsset{}
+	page := 1
+
+	for {
+		assets, res, err := a.client.Repositories.ListReleaseAssets(ctx, a.owner, a.repo, releaseID, &github.ListOptions{Page: page})
+		if err != nil {
+			return nil, err
+		}
+
+		if res.StatusCode != http.StatusOK {
+			return nil, errors.Wrapf(err, "invalid status code: %s", res.Status)
+		}
+
+		result = append(result, assets...)
+
+		if res.NextPage <= page {
+			break
+		}
+
+		page = res.NextPage
```

Comment:
Your infinite loop rely on a page number provided by the API you call. I suggest using a simple increment here to be certain that `page` increment on each iteration

Reply:
see IRL